mycpp/gc_list.h

```diff
@@ -434,11 +434,20 @@ class ListIter {
     return ret;
   }
 
+  // only for use with generators
+  List<T>* list() { return L_; }
+
  private:
   List<T>* L_;
   int i_;
 };
 
+// list(it) returns the iterator's backing list
+template <typename T>
+List<T>* list(ListIter<T> it) {
+  return list(it.list());
```

Comment:
potential optimization: This could just be `return it.list()`, but the backing lists for the iterators used for generators live on the stack. If we change `mycpp` to allocate one we could avoid this copy, but it's probably not worth the hassle... Thoughts?

Reply:
Hm isn't it just the pointer that lives on the stack?    Does anything break if you remove `list()`?

Although remember we have to be careful about rooting when we call back into the interpreter.  RootCompleter eventually does that

----

Nit: maybe call it `GetList()` or something, since `list()` is hard to search for

Reply:
> Hm isn't it just the pointer that lives on the stack? Does anything break if you remove list()?

Yeah, in most cases it's just the pointer that's on the stack, but for some reason (buried somewhere in zulip I think...) we decided to put the lists for generators on the stack in #1448

> Nit: maybe call it GetList() or something, since list() is hard to search for

👍

Reply:
Ah right :)  Yeah I suggested that to save GC pressure

But I think the reverse() can put it on the heap?

Reply:
`List::reverse()` looks like it operates on the list in place. so, we'll still need `list()`